### PR TITLE
Initial 404 page

### PR DESCRIPTION
**Pull request by adampetrie**
This is a first go at addressing https://github.com/jekyll/jekyll/issues/5039. Not sure exactly what we're looking for here so I figured I'd open a PR and use the feedback to help this take shape.

### CODE REVIEW

File: site/404.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+layout: default
+permalink: /404.html
+---
+
+<section class="intro">
+  <div class="grid">
+    <div class="unit whole center-on-mobiles">
+      <p class="first">404 - Page Not Found</p>
```
Comment (envygeeks) [inline site/404.html]: This should be generic "not found" not everything is a page.

File: site/404.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+layout: default
+permalink: /404.html
+---
```
Comment (benbalter) [inline site/404.html]: You'll want to add `sitemap: false` to prevent it from being indexed.

**Comment by adampetrie** [comment]
Here it is now:

screen shot 2016-07-27 at 7 09 58 pm

A bit more error-pagey IMO.

**Comment by benbalter** [comment]
This is terrible, but instead of "404 - not found", what about, "Huh. It seems that page is Hyde-ing..."?

**Comment by parkr** [comment]
@adampetrie *Much* better than before! I wonder if we should provide a short description of each page, e.g. `Community – Connect with other Jekyllites` etc? Giving a little insight into what's on the page would fill out the blank space here and provide a bit more help getting to the resource they are looking for.

File: site/404.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+<section class="error">
+  <div class="grid">
+    <div class="unit whole center-on-mobile">
+      <p>The resource you requested was not found. Here are some links to help you find your way:</p>
+      <ul>
+        <li>
+          <a href="/">Home</a>
+        </li>
+        <li>
+          <a href="/docs/home/">Docs</a>
```
Comment (parkr) [inline site/404.html]: I'd call this `Documentation` – it's `Docs` in the header because it's too wide 😄

**Comment by adampetrie** [comment]
I am pro puns so the 'hyde-ing' bit is awesome. There should probably be a 404 mention somewhere on the page though?

**Comment by adampetrie** [comment]
How's this? I'm just spit-ballin' here...

screen shot 2016-07-27 at 7 51 27 pm

**Comment by parkr** [comment]
LGTM. Let's ship it!

@jekyllbot: merge +site

**Pull request merged**